feat: rct225 new fellows

rct225 opened the pull request:
Hi @henryiii,

This is my attempt to add some date filtering to the fellows page.

I started by reworking the date field into an array in the YAML for the fellows markdown pages.  I used an array of "dictionaries" to overcome the issue when we have a fellow who returns (i.e. Reik).  I'm probably referring to the types wrong as far as YAML, but you can see what I did if you look at the files.

Then using this date field, I ensured the fellows page would display the date in a similar fashion to the current page.  On the main page, I created a way of limiting the list of fellows to the "current" round of fellows.  Basically I compare the end_date to the current_date minus 12 weeks - which gets us approximately into the current range of dates of the current fellows.  If end date is greater than 12 week from today, display the entry.  

I'm probably explaining this poorly, but I think this gets the page to a point where we can limit the displayed fellows by date. 

Could you take a look and let me know what you think?

rct225 committed:
Start of adding date filtering to fellows page.

rct225 committed:
Adding new date format to Fellows pages

rct225 committed:
Proposed date format for fellows on main page

rct225 committed:
Attempt to use "new" fellows dates to limit fellows listed on main page to "current" session of fellows.

matthewfeickert commented:
Converted to a draft to ensure it isn't merged accidentally.

rct225 commented:
The one thing I forgot is the individual fellows page.  I just need to update the template to fix the individual pages.

rct225 committed:
Updated fellows to use new YAML data for dates in fellows markdown files

henryiii commented on _layouts/fellow.html:
```diff
@@ -14,11 +14,9 @@
-{%- endif -%}
+{% for dates in page.dates %}
+  {{dates.start_date | date: "%b" }} - {{dates.end_date | date: "%b %Y"}}<br>
+{% endfor %}
```
I think it would nice to keep the optional array, that is 

```
{%- if page.dates.first -%}
  {% for dates in page.dates %}
    {{ dates.start_date | date: "%b" }} - {{ dates.end_date | date: "%b %Y" }}<br>
  {% endfor %}
 {%- else -%}
  {{ page.dates.start_date | date: "%b" }} - {{ page.dates.end_date | date: "%b %Y" }}<br>
 {%- endif -%}
```

That way, you can forget the "-" before `start-date`. The other parts work like this.

rct225 replied:
I see what you're saying.  Fellows will have a start_date and end date formatted as I indicate.  For cases like Reik, we then create the case with a "dates" array with pairs of start and end dates.  Then incorporate your idea to differentiate between the "single" case and the "multiple" case. I'm going to push up another revision.

rct225 committed:
New date format.

Usual case - fellow has start_date and end_date (YYYY-MM-DD)

Case of multiple fellowships - fellow has array "dates:" with multiple
pairs of start_date and end_date

rct225 committed:
Handle new date formats on individual fellows pages.

rct225 committed:
Only show "current" fellows on fellows page.  Current defined manually as "current_fellows_start"

rct225 committed:
Only show "current" fellows on main page.  Current period start is defined manually as "current_fellows_start"

rct225 committed:
A parking spot for former IRIS-HEP Fellows.  Any fellows who's start date is before the beginning of the period defined manually in "current_fellows_start"

henryiii commented:
Not quite, what I was saying was:

Allow this:

```yaml
dates:
  start: ...
  end: ...
```
_And_ this:

```yaml
dates:
- start: ...
  end: ...
```

Forgetting the dash is very easy to do. But I liked the structure of dates: ..., just not that it was forced to be a list when most fellows do not have multiple appointments.

henryiii committed:
refactor: use slighly simpler names

henryiii committed:
chore: remove whitespace

henryiii commented:
I've applied my suggestion. Other than a minor change to the logic, the bulk of the change was:

```bash
sed -I "" 's/start_date:/dates:\n  start:/g' *.md
sed -I "" 's/end_date:/  end:/g' *.md
```

(remove the `""` after the `-I` for GNU tools instead of FreeBSD-based Apple tools). Then a quick tweak to Pratush's file.

henryiii committed:
refactor: pull common code out to include

henryiii commented:
@rct225 I've rebased and added three commits (feel free to look at them individually if you want), and I think this ready to go in? Let me know.

rct225 reviewed (commented):
@henryiii I'm fine with the changes.  I'll defer to your expertise.  Thanks for your help.  Go ahead please.
